test/unittest/compiler/unittest_interpreter.cpp

```diff
@@ -45,6 +45,41 @@ const std::string pathResolver(const std::string &path) {
 auto ini_interpreter =
   std::make_shared<nntrainer::IniGraphInterpreter>(ac, pathResolver);
 
+/**
+ * @brief prototypical version of checking graph is equal
+ *
+ * @param lhs compiled(later, finalized) graph to be compared
+ * @param rhs compiled(later, finalized) graph to be compared
+ * @return true graph is equal
+ * @return false graph is not equal
+ */
+static void graphEqual(const nntrainer::GraphRepresentation &lhs,
+                       const nntrainer::GraphRepresentation &rhs) {
+  auto layers = lhs.getLayers();
+  auto ref_layers = rhs.getLayers();
+  EXPECT_EQ(layers.size(), ref_layers.size());
+
+  auto is_node_equal = [](const nntrainer::Layer &l,
+                          const nntrainer::Layer &r) {
+    nntrainer::Exporter lhs_export;
+    nntrainer::Exporter rhs_export;
+
+    l.export_to(lhs_export);
+    r.export_to(rhs_export);
+
+    /*** fixme, there is one caveat that order matters in this form */
+    EXPECT_EQ(
+      lhs_export.get_result<nntrainer::ExportMethods::METHOD_STRINGVECTOR>(),
+      rhs_export.get_result<nntrainer::ExportMethods::METHOD_STRINGVECTOR>());
+  };
```

Comment:
I wonder how we define "equal two network graphs"?

Reply:
I agree, making a graph equal have some different way to define.
For now, I don't have any idea to cover all the requirements, so didn't actually make an equal operator.

My thinking is that we could make a discussion if equal operator is needed in the future.